src/components/review/AllReviewsModal.component.js

```diff
@@ -0,0 +1,223 @@
+import React, { useState, useEffect } from "react";
+import LinearProgress from "@material-ui/core/LinearProgress";
+import moment from "moment";
+import { NavLink } from "react-router-dom";
+
+import defaultProfile from "../../../src/assets/img/default-profile.png";
+import KeyboardBackspaceIcon from "@material-ui/icons/KeyboardBackspace";
+import StarBorderIcon from "@material-ui/icons/StarBorder";
+import StarIcon from "@material-ui/icons/Star";
+import CheckBoxOutlineBlankIcon from "@material-ui/icons/CheckBoxOutlineBlank";
+import CheckBoxIcon from "@material-ui/icons/CheckBox";
+
+const AllReviewsModal = ({ reviews, closeModal, openPostReview }) => {
+  const [fiveStars, setFiveStars] = useState(0);
+  const [fiveStarsAverage, setFiveStarsAverage] = useState(0);
+  const [fourStars, setFourStars] = useState(0);
+  const [fourStarsAverage, setFourStarsAverage] = useState(0);
+  const [threeStars, setThreeStars] = useState(0);
+  const [threeStarsAverage, setThreeStarsAverage] = useState(0);
+  const [twoStars, setTwoStars] = useState(0);
+  const [twoStarsAverage, setTwoStarsAverage] = useState(0);
+  const [oneStar, setOneStar] = useState(0);
+  const [oneStarAverage, setOneStarAverage] = useState(0);
+
+  const [filter, setFilter] = useState(0);
+
+  useEffect(() => {
+    let total5 = 0;
+    let total4 = 0;
+    let total3 = 0;
+    let total2 = 0;
+    let total1 = 0;
+
+    reviews.forEach((review) => {
+      switch (review.stars) {
+        case 5:
+          total5 = total5 + 1;
+          break;
+        case 4:
+          total4 = total4 + 1;
+          break;
+        case 3:
+          total3 = total3 + 1;
+          break;
+        case 3:
+          total3 = total3 + 1;
+          break;
+        case 2:
+          total2 = total2 + 1;
+          break;
+        case 1:
+          total1 = total1 + 1;
+          break;
+        default:
+          return;
+      }
+      setFiveStars(total5);
+      setFourStars(total4);
+      setThreeStars(total3);
+      setTwoStars(total2);
+      setOneStar(total1);
+
+      setFiveStarsAverage((total5 / reviews.length) * 100);
+      setFourStarsAverage((total4 / reviews.length) * 100);
+      setThreeStarsAverage((total3 / reviews.length) * 100);
+      setTwoStarsAverage((total2 / reviews.length) * 100);
+      setOneStarAverage((total1 / reviews.length) * 100);
+    });
```

Comment:
can we add `// eslint-disable-next-line react-hooks/exhaustive-deps` after line 68 to disable the warning